assets/js/listener.js

```diff
@@ -95,3 +106,45 @@ function sendFeedback() {
     // disable the send button
     sendButton.disabled = true;
 }
+
+function switchTab(event, tabId) {
+    const container = event.target.closest('.code-tabs');
+    const tabContent = container.getElementsByClassName('tab');
+    const tabButtons = container.getElementsByClassName('tab-button');
+    
+    // Remove active class from all tabs and buttons
+    Array.from(tabContent).forEach(tab => {
+        tab.classList.remove('active');
+    });
+    
+    Array.from(tabButtons).forEach(button => {
+        button.classList.remove('active');
+    });
+    
+    // Add active class to selected tab and button
+    const selectedTab = container.querySelector(`#${tabId}`);
+    selectedTab.classList.add('active');
```

Comment:
Safer to guard this:
```suggestion
    selectedTab?.classList.add('active');
```